tools/code-owners-parser/Azure.Sdk.Tools.RetrieveCodeOwners/Program.cs

```diff
@@ -125,5 +123,24 @@ private static bool VerifyAlias(string uriStub)
                 throw;
             }
         }
+        private static IEnumerable<CodeOwnerEntry> findOwnerEntries(string target, List<CodeOwnerEntry> parsedEntries)
+        {
+            var originalTarget = target;
+            while (!String.IsNullOrEmpty(target))
+            {
+                var currentLevelEntries = parsedEntries.Where(
+                entries => entries.PathExpression.Trim(new char[] { '/', '\\' }).Equals(target.Trim(new char[] { '/', '\\' })));
+                if (currentLevelEntries.Count() > 0)
+                {
+                    return currentLevelEntries;
+                }
+                else
+                {
+                    target = target.Remove(target.LastIndexOf("/") + 1);
+                }
+            }
+
+            throw new ArgumentException(String.Format("Parameter 'target' {0} from 'targetDirectory' is not valid.", originalTarget));
```

Comment:
We probably don't want to throw an exception but instead return null or empty of some sort, as there will definitely be paths that don't match and we don't want to fail in that case here.